lldb/include/lldb/DataFormatters/TypeSynthetic.h

```diff
@@ -34,7 +34,9 @@ class SyntheticChildrenFrontEnd {
 
 public:
   SyntheticChildrenFrontEnd(ValueObject &backend)
-      : m_backend(backend), m_valid(true) {}
+      : m_backend(backend), m_valid(true) {
+    backend.SetSyntheticFrontend(this);
+  }
```

Comment:
Ignoring whether or not ValueObject should hold a pointer to a SyntheticFrontEnd, I'm concerned about the lifetime management of this solution. If a SyntheticChildrenFrontEnd is destroyed, the underlying `ValueObject` will be holding a dangling pointer.